tenets/codelingo/go/unused-private-const-var-same-package/codelingo.yaml

```diff
@@ -0,0 +1,32 @@
+tenets:
+  - name: unused-private-const-var-same-package
+    doc: |
+      Find private const or var which are unused in a given package.
+    flows:
+      codelingo/review:
+        comment: |
+          '{{thingname}}' is private yet it is not referenced anywhere else in its package.
+          It should probably be removed.
+    query: |
+      import codelingo/ast/go
+
+      go.dir:
+        go.file(depth = any):
+          @ review.vars.myfname
+          filename
+          go.decls:
+            go.gen_decl:
+              go.value_spec:
+                go.names:
+                  @ review.comment
+                  go.ident:
+                    @ review.vars.thingname
+                    name as privateVarName
+                    private == "true"
+        exclude:
+          go.file(depth = any):
+            go.call_expr(depth = any):
+              go.args:
+                go.ident:
+                  name == privateVarName
+                  private == "true"
```

Comment:
I think you can shorten it down. Does this work?

```
go.dir:
  go.value_spec(depth = any):
    go.names:
      @ review.comment
      go.ident:
        @ review.vars.thingname
        name as privateVarName
        private == "true"
  exclude:
    go.call_expr(depth = any):
      go.args:
        go.ident:
          name == privateVarName
          private == "true"
```

Reply:
This is geting slightly different behavior. It also appears to have matched one of the structs, which may have been a good thing if I were attempting to combine the 3 tenets, but I'm separating the logic.

Reply:
Oh interesting. I guess that the `go.gen_decl` fact is what defines a variable/const definition. Can you at least remove the `go.file` facts without changing the output?

Reply:
You can probably also remove the `go.decls` fact.

Reply:
(this isn't blocking btw, it's just nice to get them concise)

Reply:
I will give it a go.

Reply:
Sweet as. Getting insight as to which facts exactly are unique to a piece of syntax is what I'm hoping to achieve by making my CLQL a little verbose. Please bear with me :)